src/mrdParser.ts

```diff
@@ -2,78 +2,64 @@ import * as fs from 'fs';
 
 import { Participant } from './sequintence/participant';
 import { Relation } from './sequintence/relation';
+import { Diagram } from './types/diagram.type';
 
-export class MrdParser {
-  private participants: Participant[];
-  private relations: Relation[];
+type sourceAndTargetParticipants = {
+  sourceName: string;
+  targetName: string;
+};
 
-  constructor(path: string) {
-    this.parseMrd(path);
-  }
-
-  private getSourceAndTargetParticipantNames(string) {
-    let relationDirection = string.substring(string.indexOf(`(`) + 1, string.lastIndexOf(`)`));
+export function mrdParser(pathToFile: string): Diagram {
+  const participants = [];
+  const relations = [];
 
-    const splittedDirection = relationDirection.split('>>');
-    const sourceName = splittedDirection[0].trim();
-    const targetName = splittedDirection.pop().trim();
+  const fileContent = fs.readFileSync(pathToFile, 'utf-8');
+  const rows = fileContent.split(`\n`);
+  const [diagramType, ...diagramBody] = rows.map((e) => e.trim());
 
-    return {
-      sourceName,
-      targetName,
-    };
+  if (diagramType != 'sequenceDiagram') {
+    throw new Error('unimplemented diagram format');
   }
 
-  private getParticipantByName(name) {
-    const foundedParticipant = this.participants.find((e) => e.name === name);
+  diagramBody.forEach((e, i) => {
+    let splittedString = e.split(':');
```

Comment:
Please consider using destructuring here
```suggestion
    let [nodeType, nodeValue] = e.split(':');
    nodeType = nodeType.trim();
```